### DISCUSSION TITLE
EnhancedAPI Refresh

### DISCUSSION THREAD

**Discussion by muhssamy** [TabularEditor/TabularEditor3#1398] [Ideas]
it will be great if tabular editor 3 has a feature that can post enhanced Api calls for the service and track its status. this will help a lot in managing large semantic models

**Comment by otykier** [comment]
I’m not sure what you’re referring to. Where do you feel the current XMLA refresh option falls short?

**Reply by muhssamy** [reply]
i am referring to this 
https://learn.microsoft.com/en-us/power-bi/connect-data/asynchronous-refresh
same here https://learn.microsoft.com/en-us/rest/api/power-bi/datasets/refresh-dataset

**Where do you feel the current XMLA refresh option falls short? **
1. batch commit
2. ignore refreshpolicy if needed 
3. control the parallel degree
4. this is not requiring my client to be connected to the server all the time of refresh. just you post the command for refresh 
then follow it up by other calls if needed. this really helps in large models

**Reply by otykier** [reply]
Technically speaking, your points #&#8203;1, #2 and #3 apply to XMLA refreshing as well - it's just a matter of executing the proper TMSL command. However, point #4 is valid as XMLA refreshes require the connection to remain active for the duration of the refresh.

In general, though, I wouldn't recommend using TE3 (or any other client tool, for that matter), to perform refresh operations that are known to take a long time (say, 1 hour or more) to complete. Such operations are just as easily started from within the Power BI service. The value of being able to refresh the model from TE3, in my opinion, lies mostly within quick refreshes, for example to import a small amount of test data, or to recalculate the model after making changes to relationships, calculated tables, calculated columns, or calculation groups.

That being said, we'll consider our options. We may be able to deliver something that addresses #&#8203;1, #2 and #3 through the existing XMLA refresh mechanism in the shorter term. But #4 is something we'll have to consider based on popular demand. I don't see it being very important for the reason mentioned above, but also for the fact that not all users who have XMLA access, may have the required permissions to connect to the REST APIs. Moreover, this feature would not bring any value to SSAS / AzureAS users, which represent a considerable portion of our customers.

**Reply by muhssamy** [reply]
Thanks

**Reply by edhans** [reply]
Another data point - you cannot use the service (that I am aware of) to refresh partitions or individual tables, some of which may require more than 1 hour in really large models. I still use TE/3 for quick refreshes, but for larger models where I need to refresh historical partitions, I use VS Code with the Power BI Studio plugin where I can kick off the refresh then log off. It uses the enhanced API method.

**Reply by muhssamy** [reply]
@otykier thanks i can see it is now available in advanced refresh i know it is `XMLA` refresh. but it will be very helpful in certain scenarios

**Comment by mlonsk** [comment]
The functionality is available in Tabualar Editor 3.25.0 and onwards in the advanced refesh dialog. See [our documentation](https://docs.tabulareditor.com/features/advanced-refresh.html) for more info.
